skyfield/iokit.py

```diff
@@ -350,9 +358,14 @@ def parse_leap_seconds(fileobj):
 def parse_celestrak_tle(fileobj):
     lines = iter(fileobj)
     for line in lines:
-        name = line.decode('ascii').strip()
-        line1 = next(lines).decode('ascii')
-        line2 = next(lines).decode('ascii')
+        if line.decode('ascii').strip()[0] == '0':
+            name = line.decode('ascii').strip()
+            line1 = next(lines).decode('ascii')
+            line2 = next(lines).decode('ascii')
+        else:  # two-line element set, no name provided!
+            line1 = line.decode('ascii')
+            line2 = next(lines).decode('ascii')
+            name = line2.split()[1]  # grab satellite ID# as name
```

Comment:
Instead of doing a `split()` here that repeats the work that `EarthSatellite` will do to parse the name, let's try pivoting to `name = None` here and then set the name after the satellite object has been created by pulling its own opinion of the satellite ID that it will have parsed out from the input lines, so that we don't do the same work twice. Thanks!

Reply:
Will do! I just have to figure out the right way to access and set these.